backends/bmv2/common/control.cpp

```diff
@@ -71,7 +71,7 @@ void ControlConverter::convertTableEntries(const IR::P4Table *table,
                     key->emplace("key", stringRepr(km->left->to<IR::Constant>()->value, k8));
                     auto trailing_zeros = [](unsigned long n) { return n ? __builtin_ctzl(n) : 0; };
                     auto count_ones = [](unsigned long n) { return n ? __builtin_popcountl(n) : 0;};
-                    unsigned long mask = km->right->to<IR::Constant>()->value.get_ui();
+                    auto mask = static_cast<unsigned long>(km->right->to<IR::Constant>()->value);
```

Comment:
Why are static casts needed? Can't this operation be done directly on the big_int object?

Reply:
This one could be done directly on big_int (and could have been done directly on mpz_class before), I was just minimalizing the changes.